framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -203,20 +201,31 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		// Save blockHeader, aggregateCommit, validatorsData and cross chain messages if any.
 		try {
 			// Fetch last certificate from the receiving chain and update the _lastCertificate
-			chainAccountJSON = await this._receivingChainClient.invoke<ChainAccountJSON>(
-				'interoperability_getChainAccount',
-				{ chainID: this._ownChainID.toString('hex') },
-			);
-
-			// If sending chain is not registered with the receiving chain then only save data on new block and exit
-			if (!chainAccountJSON || (chainAccountJSON && !chainAccountJSON.lastCertificate)) {
-				this.logger.info(
-					'Sending chain is not registered to the receiving chain yet and has no chain data.',
+			try {
+				chainAccountJSON = await this._receivingChainClient.invoke<ChainAccountJSON>(
+					'interoperability_getChainAccount',
+					{ chainID: this._ownChainID.toString('hex') },
 				);
+				// If sending chain is not registered with the receiving chain then only save data on new block and exit
+				if (!chainAccountJSON || (chainAccountJSON && !chainAccountJSON.lastCertificate)) {
+					this.logger.info(
+						'Sending chain is not registered to the receiving chain yet and has no chain data.',
+					);
+					await this._saveDataOnNewBlock(newBlockHeader);
+
+					return;
+				}
+			} catch (error) {
+				// If receivingChainAPIClient is not ready then still save data on new block
 				await this._saveDataOnNewBlock(newBlockHeader);
+				this.logger.error(
+					error,
+					'Error occured while using accessing receivingChain API Client but all data is saved on newBlock.',
```

Comment:
```suggestion
					'Error occurred while using accessing receivingChain API Client but all data is saved on newBlock.',
```

Reply:
Is it guaranteed that the data was saved before the error occurred?

Reply:
it will throw error and will not log this if something goes wrong while save data